src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-notification.ts

```diff
@@ -1,5 +1,20 @@
+import { ToastOptions } from 'ng2-toastr';
 import { NotificationType } from '../enum/notification-type.enum';
 
+export class CdNotificationConfig {
+  type: NotificationType;
+  title: string;
+  message?: string; // Use this for error notifications only
+  options?: any | ToastOptions;
+
+  constructor(type: NotificationType, title: string, message?: string, options?: any) {
```

Comment:
Just suggestion on using TS syntactic sugar for this: `constructor(private type: NotificationType...)`? and skipping the previous declaration?

Reply:
done :+1: